docs/source/enterprise-guide/change-management.mdx

```diff
@@ -0,0 +1,537 @@
+
+```graphql:title=After
+# monolith — extends the User type
+extend type User @key(fields: "id") {
+  id: ID! @external
+  fieldFromMonolith: String
+}
+# microservice — "owns" the User type
+type User @key(fields: "id") {
+  id: ID!
+  fieldFromMicroservice: String
+}
+```
+
+### Release steps
+
+The simplest release process works as follows:
+
+1. **In the monolith repository**, commit a changeset containing the schema
+   change and corresponding implementation to the release branch, then trigger
+   the release pipeline.
+2. The release pipeline executes: building artifacts, publishing subgraph SDL,
+   and triggering a rolling deploy of the monolith.
+
+   At this point, the gateway continues to use the old query plans.
+   **Composition is broken**, so Apollo Studio does not publish the new
+   supergraph to the gateway.
+
+3. **In the microservice repository**, commit a changeset containing the schema
+   change and corresponding implementation to the release branch, then trigger
+   the release pipeline.
+4. The release pipeline executes: building artifacts, publishing subgraph SDL,
+   and triggering a rolling deploy of the microservice.
+
+   As soon as the subgraph SDL is publishes, composition resolves and gateways
+   are updated. They will now create query plans that use the microservice as
+   the owner of the entity.
+
+However, in the event of an issue with the new query plans, rolling this change
+back requires re-publishing the subgraph schemas, which might be a
+time-consuming process.
+
+### Release alternative: out-of-band schema publishing
+
+Because a query planning change should not require changes to our runtime code,
+we could choose to publish schemas _without_ deploying our services. The process
+would be:
+
+1. Edit the subgraph schemas for both the monolith and microservice.
+2. Manually publish them using [Rover][rover], one after the other, updating the
+   gateway after the second publish.
+3. Commit the changes to the respective repositories. The next release will be a
+   no-op because the Apollo Studio already has these changes.
+
+[rover]: https://www.apollographql.com/docs/rover/
+
+In the event of an issue, we can quickly publish the previous versions of the
+subgraph schemas to "roll back".
+
+We might choose to keep a copy of subgraph schemas in a separate code repository
+for review and auditing purposes.
+
+### Progressive delivery techniques
+
+The out-of-band schema publishing technique pairs well with a version of
+blue-green deploys. If we run two pools of gateway servers configured to load
+supergraph schemas using different variants (i.e., "prod-green" and
+"prod-blue"), we can publish our manually-edited subgraph schemas to the "blue"
+variant while the "green" variant takes all the traffic. After the launch
+completes, we can flip traffic to the "blue" variant for the query plan changes
+to take effect.
+
+Rolling back the changes is simply a matter of flipping traffic to the "green"
+variant.
```

Comment:
Yeaaah ... I tried my hardest to figure out a method for supporting blue-green schema changes, but you're absolutely right that it would fall apart with too many independently moving subgraphs. I'll remove it.